agent/api/task/task_windows.go

```diff
@@ -217,3 +183,72 @@ func (task *Task) GetCredentialSpecResource() ([]taskresource.TaskResource, bool
 func enableIPv6SysctlSetting(hostConfig *dockercontainer.HostConfig) {
 	return
 }
+
+// requiresFSxWindowsFileServerResource returns true if at least one volume in the task
+// is of type 'fsxWindowsFileServer'
+func (task *Task) requiresFSxWindowsFileServerResource() bool {
+	for _, volume := range task.Volumes {
+		if volume.Type == FSxWindowsFileServerVolumeType {
+			return true
+		}
+	}
+	return false
+}
+
+// initializeFSxWindowsFileServerResource builds the resource dependency map for the fsxwindowsfileserver resource
+func (task *Task) initializeFSxWindowsFileServerResource(cfg *config.Config, credentialsManager credentials.Manager,
+	resourceFields *taskresource.ResourceFields) error {
+	for i, vol := range task.Volumes {
+		if vol.Type != FSxWindowsFileServerVolumeType {
+			continue
+		}
+
+		fsxWindowsFileServerVol, ok := vol.Volume.(*fsxwindowsfileserver.FSxWindowsFileServerVolumeConfig)
+		if !ok {
+			return errors.New("task volume: volume configuration does not match the type 'fsxWindowsFileServer'")
+		}
+
+		err := task.addFSxWindowsFileServerResource(cfg, credentialsManager, resourceFields, &task.Volumes[i], fsxWindowsFileServerVol)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// addFSxWindowsFileServerResource creates a fsxwindowsfileserver resource for every fsxwindowsfileserver task volume
+// and updates container dependency
+func (task *Task) addFSxWindowsFileServerResource(
+	cfg *config.Config,
+	credentialsManager credentials.Manager,
+	resourceFields *taskresource.ResourceFields,
+	vol *TaskVolume,
+	fsxWindowsFileServerVol *fsxwindowsfileserver.FSxWindowsFileServerVolumeConfig,
+) error {
+	hostPath, err := utils.FindUnusedDriveLetter()
```

Comment:
Can't we just mount the FSx file share to a local folder path instead of using a drive letter which actually enforces restriction on the number of mount volumes ?

Reply:
we cannot mount fsx to a local folder path. It requires a drive letter.

Reply:
https://docs.microsoft.com/en-us/windows-server/storage/disk-management/assign-a-mount-point-folder-path-to-a-drive

Reply:
```New-SmbGlobalMapping``` powershell cmdlet enforces using drive letter for ```-LocalPath``` param.